### DISCUSSION TITLE
Pull request for ESPhome

### DISCUSSION THREAD

**Discussion by Rvh91** [stas-sl/esphome-sound-level-meter#8] [General]
Hi! I was wondering if you considered making a pull request for ESPhome? This repository is awesome! 
Able to make a reliable sound monitor for low cost. This solution is so far superior to using something like a KY-038 mic board or similar.

**Comment by stas-sl** [comment]
Hi, I'm glad you found it useful )

Considering PR into ESPHome it might be more difficult than it seems on the first glance. There were some [long discussions](https://github.com/esphome/feature-requests/issues/1254) previously how to design/structure i2s/audio components, and while some basic i2s support was added to ESPHome, there are still many open questions. You can also find my related comments [here](https://github.com/stas-sl/esphome-sound-level-meter/discussions/2) and [there](https://github.com/stas-sl/esphome-sound-level-meter/issues/5).
